core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/OutputTargetBuildItem.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.deployment.pkg.builditem;
+
+import java.nio.file.Path;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+/**
+ * The location that output artifacts should be created in
+ *
+ * TODO: should we just create them in temp directories, and leave it up to the integration to move them where they want?
+ */
+public final class OutputTargetBuildItem extends SimpleBuildItem {
```

Comment:
Maybe this should be config as well?

Reply:
The output directory? That's generally not something the user wants to manually configure as tools have their own standard locations? Or do you mean something else?

Reply:
I think it makes sense not from the perspective of a user trying to manually configure things, but from the perspective that all build steps should be configured in a consistent way. 
Sometimes a given build system might have its own configuration items which overlap with Quarkus's, but in these cases it just means the build system can provide default values for these items.  It may transpire that there are some items which are configurable in build system A that are not configurable in build system B.

And, if you make it a configuration item, then the build systems can "stuff" it with a default as needed like we already do for application name and version, which is perhaps easier than producing a bunch of initial items which are in the end just configuration.

Reply:
Configuration is more of an end user thing though that a build system thing. Having what are basically internals exposed as user config does not seem like a great option from a user experience point of view.

Reply:
Perhaps for the case when we launch the build/provisioning outside of Maven/Gradle context. This is what creator is supporting currently, where you can configure a general work/output dir and then per phase, if necessary.

Reply:
On the contrary, it's irritating to users when they can't adjust a knob they want to adjust.  It's only "internals" when you look at the build system + augmentation combination as a single closed box; if you look at them as two separate boxes it makes a lot more sense.  As we've said from day one, it doesn't make sense to be married into a particular build system.  But all this strong coupling really prevents us from being truly portable across build systems.

I think the stronger principle is to unify all configuration as configuration.  Let the build system provide all the defaults that make sense for that particular system, and don't make any assumptions about what the build system might provide.

Reply:
I will leave this for now, and revisit it once the main functionality is done (unless I run into similar issues elsewhere).

One other use case to think about is how we generate the example config files, which is done by running a minimal build. The task needs to know the location to create the example file, which IMHO should be a build item as it is a task specific configuration piece rather than general quarkus config.

I guess the more general class of this problem is the idea that if there are configuration options that only apply to certain systems then these will still be cluttering up the config namespace of other build systems.